### PR TITLE
[BUG](es_catalog)Solve the problem of querying es catalog Unexpected exception: Index:…

### PR DESCRIPTION
**Pull request by caoliang-web**
## Problem summary

1.Solve the problem of querying es catalog Unexpected exception: Index: 0, Size: 0

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java [hunk 1]
```diff
@@ -257,11 +257,13 @@ private List<TScanRangeLocations> getShardLocations() throws UserException {
                 // Generate on es scan range
                 TEsScanRange esScanRange = new TEsScanRange();
                 esScanRange.setEsHosts(shardAllocations);
-                esScanRange.setIndex(shardRouting.get(0).getIndexName());
+                if (shardRouting.size() > 0 && shardRouting != null) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:260]: Will this cause errors in query?

Comment (caoliang-web) [reply fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:260]: The query will appear: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:260]: I mean is whether not to set index will cause subsequent query failure.

Comment (caoliang-web) [reply fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:260]: I confirm

**Comment by stalary** [comment]
run build all

**Review by stalary — APPROVED** [review]
LGTM

**Comment by caoliang-web** [comment]
run build all

**Comment by caoliang-web** [comment]
run buildall

**Comment by caoliang-web** [comment]
run build all

**Review by hf200012 — APPROVED** [review]

**Comment by caoliang-web** [comment]
run buildall

**Comment by caoliang-web** [comment]
run clickbench

**Comment by caoliang-web** [comment]
run buildall

**Comment by caoliang-web** [comment]
run buildall

**Comment by caoliang-web** [comment]
run P0

**Comment by caoliang-web** [comment]
run buildall

**Comment by caoliang-web** [comment]
run P1

**Comment by caoliang-web** [comment]
run buildall

**Pull request merged**